### ISSUE TITLE
armor.config.level_multiplier gets (hard) set down to 0.9 by shields

### ISSUE THREAD

**Issue by sirrobzeroone** [minetest-mods/3d_armor#30]
I noticed a discrepency between what the armor.config.level_multiplier is set to under 3d_armor and what the game reported level as being when you look at the armor_ui inside the game

eg diamond chestplate level should be 20 (fleshy=20), level*armor.config.level_multiplier (default=1).  However inside the game when you just wear the chestplate it reports as level 18 with shields enabled

So tracked this back through to Shields which has a block of code:
```
if minetest.global_exists("armor") and armor.elements then
	table.insert(armor.elements, "shield")
	local mult = armor.config.level_multiplier or 1
	armor.config.level_multiplier = mult * 0.9
end
```

When enabled the shields multipler of 0.9 overrides the base armor.config.level_multiplier for all armor items not just shields and there is no way to reset it back to 1...well unless you set the setting in armor_3d to 1.111etc. Which is a tough track down to get to root cause lol.

This dosen't quiet seem like the correct behaviour? I'm guessing the intention was to help with balancing? Im not sure? 

I'm not sure the armor.config.level_multiplier should be redefined in the shields module but rather leave that out and let people adjust via the base 3d_armor setting "level_multiplier" if they feel they need it? 

Took me a fair bit of time to track down what was happening, spotted it in the end using debug were you can clearly see shields suddenly adjust to 0.9 instead of 1 (this overrides for all items later when calulating in game) :

```
2020-12-13 20:07:43: [Main]: 3d_armor:helmet_admin levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:chestplate_admin levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:leggings_admin levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:boots_admin levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:helmet_wood levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:chestplate_wood levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:leggings_wood levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:boots_wood levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:helmet_cactus levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:chestplate_cactus levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:leggings_cactus levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:boots_cactus levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:helmet_steel levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:chestplate_steel levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:leggings_steel levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:boots_steel levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:helmet_bronze levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:chestplate_bronze levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:leggings_bronze levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:boots_bronze levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:helmet_diamond levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:chestplate_diamond levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:leggings_diamond levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:boots_diamond levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:helmet_gold levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:chestplate_gold levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:leggings_gold levelmulti1: 1
2020-12-13 20:07:43: [Main]: 3d_armor:boots_gold levelmulti1: 1
2020-12-13 20:07:43: [Main]: shields:shield_admin levelmulti1: 0.9
2020-12-13 20:07:43: [Main]: shields:shield_wood levelmulti1: 0.9
2020-12-13 20:07:43: [Main]: shields:shield_enhanced_wood levelmulti1: 0.9
2020-12-13 20:07:43: [Main]: shields:shield_cactus levelmulti1: 0.9
2020-12-13 20:07:43: [Main]: shields:shield_enhanced_cactus levelmulti1: 0.9
2020-12-13 20:07:43: [Main]: shields:shield_steel levelmulti1: 0.9
2020-12-13 20:07:43: [Main]: shields:shield_bronze levelmulti1: 0.9
2020-12-13 20:07:43: [Main]: shields:shield_diamond levelmulti1: 0.9
2020-12-13 20:07:43: [Main]: shields:shield_gold levelmulti1: 0.9
```

Anyways as Im not sure if this is a bug or by design? Happy to be corrected just seemed like strange behaviour that was not very appaarent to the end user...unless I missed some doco somewhere.

Fix is a two liner
```
if minetest.global_exists("armor") and armor.elements then
	table.insert(armor.elements, "shield")
end
```
full set diamond no shields enabled, level= 77 (w/ fullset 10% bonus)
full set diamond plus shields without fix, level= 84.5 (w/ fullset 10% bonus)
full set diamond plus shields with fix, level = 93.5 (w/ fullset 10% bonus)

**Comment by BuckarooBanzay** [comment]
This smells like a workaround of some kind :thinking: 
Are you sure the fix does not introduce any more bugs?

**Comment by sirrobzeroone** [comment]
Not that I can tell and Im not even sure why Stu has setup shields with 0.9 multiplyier. It's intresting because if you register a later mod eg my gloves/gauntlets. You can illicit even more intresting behaviour, say you set gloves to 0.9 as well (as I had as I copied shields and didnt fully understand those lines) you end up with 1*0.9*0.9 = 0.81 multiplyier which is even stranger when you view the armor UI in game. 

I reset all three of the above mods to basically respect 3d_armor multi only (ie remove those 2 lines for both). This now pushes my armor upto 110 if I use all 6 pieces @ diamond. 

I have had no issues with mobs I need to check pvp out more fully, but so far no issue. I'm honestly not expecting to see any as Stu designed the multiplier to go up and down to make armor more/less effective. Just beyond 100 gets a bit redundant for fleshy if Im understanding more fully how 3d_armor works. 

I more put this here as a placeholder as I investigate pull things apart a bit more and see if I notice any oddities. But thought if I include the fix a few others might check it out by manually removing those two lines (okay hoped) and see if they get any issues. More eyes less chance of missing something.

I'd like to push that fix for issue #13 through first as its a much more clear-case fix that something is wrong. Where as the above is much more um nerve-racking at least for myself.

**Comment by BuckarooBanzay** [comment]
fixed by https://github.com/minetest-mods/3d_armor/commit/5642819ba5edb7da8ae68537121fbb2bfc5ec3f8
